pkg/completion/completion.go

```diff
@@ -24,3 +31,49 @@ func AutocompleteNone(cmd *cobra.Command, args []string, toComplete string) ([]s
 func AutocompleteDefault(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
 	return nil, cobra.ShellCompDirectiveDefault
 }
+
+// AutocompleteCapabilities - Autocomplete linux capabilities options.
+// Used by --cap-add and --cap-drop.
+func AutocompleteCapabilities(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	caps := capabilities.AllCapabilities()
+	// trim "CAP_" if the user doesn't type CA...
+	if !strings.HasPrefix(toComplete, "CA") {
```

Comment:
Why is this needed.  TrimPrefix will be a noop if CAP_ does not exist?

Reply:
The capabilities list contains all capabilities with the `CAP_` prefix. But I want to suggest them without this prefix. If a user already started typing `CA` then I want to complete them with the prefix. We currently allow both `CHOWN` and `CAP_CHOWN` as valid argument.

Reply:
Do we need to check if toComplete is lowercase or uppercase?
Instead of trimming prefix of caps, can we add cap_ prefix to `toComplete` here?

Reply:
Changing `toComplete` doesn't work. The shell will filter the result on the typed value. I cannot change this value.

That is how it behaves with the current implementation:
```
$ podman run --cap-add [TAB]
ALL               BLOCK_SUSPEND     FOWNER            KILL              MAC_OVERRIDE      NET_BROADCAST     SETPCAP           SYS_CHROOT        SYS_PACCT         SYS_TIME
AUDIT_CONTROL     CHOWN             FSETID            LEASE             MKNOD             NET_RAW           SETUID            SYSLOG            SYS_PTRACE        SYS_TTY_CONFIG
AUDIT_READ        DAC_OVERRIDE      IPC_LOCK          LINUX_IMMUTABLE   NET_ADMIN         SETFCAP           SYS_ADMIN         SYS_MODULE        SYS_RAWIO         WAKE_ALARM
AUDIT_WRITE       DAC_READ_SEARCH   IPC_OWNER         MAC_ADMIN         NET_BIND_SERVICE  SETGID            SYS_BOOT          SYS_NICE          SYS_RESOURCE
$ podman run --cap-add C[TAB]    <-- This will become CHOWN as only one capability start with C
$ podman run --cap-add CA[TAB]   <-- This will become CAP_
$ podman run --cap-add CAP_[TAB]
CAP_AUDIT_CONTROL     CAP_DAC_OVERRIDE      CAP_IPC_OWNER         CAP_MAC_OVERRIDE      CAP_NET_RAW           CAP_SYS_ADMIN         CAP_SYS_NICE          CAP_SYS_TIME
CAP_AUDIT_READ        CAP_DAC_READ_SEARCH   CAP_KILL              CAP_MKNOD             CAP_SETFCAP           CAP_SYS_BOOT          CAP_SYS_PACCT         CAP_SYS_TTY_CONFIG
CAP_AUDIT_WRITE       CAP_FOWNER            CAP_LEASE             CAP_NET_ADMIN         CAP_SETGID            CAP_SYS_CHROOT        CAP_SYS_PTRACE        CAP_WAKE_ALARM
CAP_BLOCK_SUSPEND     CAP_FSETID            CAP_LINUX_IMMUTABLE   CAP_NET_BIND_SERVICE  CAP_SETPCAP           CAP_SYSLOG            CAP_SYS_RAWIO         
CAP_CHOWN             CAP_IPC_LOCK          CAP_MAC_ADMIN         CAP_NET_BROADCAST     CAP_SETUID            CAP_SYS_MODULE        CAP_SYS_RESOURCE
```

This only works for uppercase but I could add support for lowercase if this is wanted.

Reply:
Thanks for the explanation.

Reply:
I think we want upper and lower case.  But good explanation.

podman run --cap-add m<tab> would be nice to expand to mknod

Reply:
Done.

```
paul ~/go/src/github.com/containers/libpod ✓ $ bin/podman run --cap-add cap_net_
cap_net_admin         cap_net_bind_service  cap_net_broadcast     cap_net_raw           
paul ~/go/src/github.com/containers/libpod ✓ $ bin/podman run --cap-add net_
net_admin         net_bind_service  net_broadcast     net_raw           
paul ~/go/src/github.com/containers/libpod ✓ $ bin/podman run --cap-add NET_
NET_ADMIN         NET_BIND_SERVICE  NET_BROADCAST     NET_RAW           
paul ~/go/src/github.com/containers/libpod ✓ $ bin/podman run --cap-add CAP_
CAP_AUDIT_CONTROL     CAP_DAC_OVERRIDE      CAP_IPC_OWNER         CAP_MAC_OVERRIDE      CAP_NET_RAW           CAP_SYS_ADMIN         CAP_SYS_NICE          CAP_SYS_TIME
CAP_AUDIT_READ        CAP_DAC_READ_SEARCH   CAP_KILL              CAP_MKNOD             CAP_SETFCAP           CAP_SYS_BOOT          CAP_SYS_PACCT         CAP_SYS_TTY_CONFIG
CAP_AUDIT_WRITE       CAP_FOWNER            CAP_LEASE             CAP_NET_ADMIN         CAP_SETGID            CAP_SYS_CHROOT        CAP_SYS_PTRACE        CAP_WAKE_ALARM
CAP_BLOCK_SUSPEND     CAP_FSETID            CAP_LINUX_IMMUTABLE   CAP_NET_BIND_SERVICE  CAP_SETPCAP           CAP_SYSLOG            CAP_SYS_RAWIO         
CAP_CHOWN             CAP_IPC_LOCK          CAP_MAC_ADMIN         CAP_NET_BROADCAST     CAP_SETUID            CAP_SYS_MODULE        CAP_SYS_RESOURCE      
paul ~/go/src/github.com/containers/libpod ✓ $ bin/podman run --cap-add CAP_NET_
CAP_NET_ADMIN         CAP_NET_BIND_SERVICE  CAP_NET_BROADCAST     CAP_NET_RAW           
paul ~/go/src/github.com/containers/libpod ✓ $ bin/podman run --cap-add cap_
cap_audit_control     cap_dac_override      cap_ipc_owner         cap_mac_override      cap_net_raw           cap_sys_admin         cap_sys_nice          cap_sys_time
cap_audit_read        cap_dac_read_search   cap_kill              cap_mknod             cap_setfcap           cap_sys_boot          cap_sys_pacct         cap_sys_tty_config
cap_audit_write       cap_fowner            cap_lease             cap_net_admin         cap_setgid            cap_sys_chroot        cap_sys_ptrace        cap_wake_alarm
cap_block_suspend     cap_fsetid            cap_linux_immutable   cap_net_bind_service  cap_setpcap           cap_syslog            cap_sys_rawio         
cap_chown             cap_ipc_lock          cap_mac_admin         cap_net_broadcast     cap_setuid            cap_sys_module        cap_sys_resource      
paul ~/go/src/github.com/containers/libpod ✓ $ bin/podman run --cap-add m
mac_admin     mac_override  mknod         
paul ~/go/src/github.com/containers/libpod ✓ $ bin/podman run --cap-add M
MAC_ADMIN     MAC_OVERRIDE  MKNOD         
paul ~/go/src/github.com/containers/libpod ✓ $ bin/podman run --cap-add 
ALL               BLOCK_SUSPEND     FOWNER            KILL              MAC_OVERRIDE      NET_BROADCAST     SETPCAP           SYS_CHROOT        SYS_PACCT         SYS_TIME
AUDIT_CONTROL     CHOWN             FSETID            LEASE             MKNOD             NET_RAW           SETUID            SYSLOG            SYS_PTRACE        SYS_TTY_CONFIG
AUDIT_READ        DAC_OVERRIDE      IPC_LOCK          LINUX_IMMUTABLE   NET_ADMIN         SETFCAP           SYS_ADMIN         SYS_MODULE        SYS_RAWIO         WAKE_ALARM
AUDIT_WRITE       DAC_READ_SEARCH   IPC_OWNER         MAC_ADMIN         NET_BIND_SERVICE  SETGID            SYS_BOOT          SYS_NICE          SYS_RESOURCE 
```